cmd/bnscli/common.go

```diff
@@ -1,31 +1,75 @@
 package main
 
 import (
+	"encoding/base64"
 	"encoding/binary"
+	"encoding/hex"
 	"errors"
+	"fmt"
 	"io"
 	"os"
+	"strconv"
 
 	"github.com/iov-one/weave/cmd/bnsd/app"
 )
 
+// unpackSequence process given raw string representation and does its best in
+// order to decode a sequence value from the raw form. This function is
+// intended to be used with data coming from stdin.
+//
+// Supported formats are:
+// - string encoded decimal number
+// - hex encoded binary sequence value
+// - base64 encoded binary  sequence value
+func unpackSequence(raw string) ([]byte, error) {
```

Comment:
Wow, quite adaptable parser here.

I like this

Reply:
This could be pulled into `weave` and used to declare a `Sequence` type, similar to `weave.UnixTime` or `weave.Address`.

The current implementation [is using `[]byte`](https://github.com/iov-one/weave/blob/master/x/escrow/codec.pb.go#L347). This provides no documentation and results in [an obfuscated JSON representation](https://github.com/iov-one/weave/blob/master/cmd/bnscli/clitests/batch_proposal.test.gold#L26). Using a custom type would provide an extra type check, `Validate()` method and a human readable JSON representation.